### PR TITLE
feat: Allow classifications based on metadata properties, add PDF and plain text providers

### PR DESCRIPTION
**Pull request by marcelklehr**
image

### CODE REVIEW

**Comment by Rello** [comment]
related to the screenshot, the current "... if" I think are confusing. are the filters linked with or or and?

**Comment by marcelklehr** [comment]
They're linked with "or"

**Comment by Rello** [comment]
what do you think to use this PR to correct this?

**Comment by marcelklehr** [comment]
yep, will add the word "or" to these lines. It's the weekend now for me ;) Back at it on monday

File: src/components/ClassificationLabel.vue [hunk 1]
```diff
@@ -36,9 +36,29 @@
...
 				</label>
 			</div>
+			<div class="option metadata">
+				<label>
+					<span class="text">{{ t('files_confidential', '… or if document has all metadata values') }}</span>
+					<template v-for="(item, key) in label.metadataItems">
+						<NcTextField :key="'key' + key"
+							class="field"
+							:value.sync="item.key"
+							:title="item.key"
```
Comment (julien-nc) [inline src/components/ClassificationLabel.vue:46]: Is it intended that the value is the same as the title?

**Review by julien-nc — COMMENTED** [review]
Minor remarks. LGTM. Didn't try it though.

File: src/components/ClassificationLabel.vue [hunk 2]
```diff
@@ -36,9 +36,29 @@
...
+							label="Metadata key"
```
[the remark this answers is not in the document]
Comment (marcelklehr) [reply src/components/ClassificationLabel.vue:47]: Oh, yes

**Review by julien-nc — APPROVED** [review]

**Pull request merged**